components/Controllers/OpenVault/VaultWriteButton.tsx

```diff
@@ -143,8 +145,27 @@ export function VaultWriteButton({
 
   const buttonText = useMemo(() => {
     if (!oracleInfoSynced) return 'Waiting for oracle...';
-    return vaultHasDebt ? 'Update Loan' : 'Borrow';
-  }, [oracleInfoSynced, vaultHasDebt]);
+    const updateOrBorrowText =
+      vaultHasDebt || vaultCollateralCount > 0 ? 'Update Loan' : 'Borrow';
+    if (amount.isZero()) {
+      if (depositNFTs.length > 0) {
```

Comment:
if they are depositing AND withdrawing we should say "Update Loan"

Reply:
yeah you're right